public/app/plugins/datasource/loki/language_provider.ts

```diff
@@ -236,13 +238,24 @@ export default class LokiLanguageProvider extends LanguageProvider {
       this.logLabelOptions = labelKeys.map(key => ({ label: key, value: key, isLeaf: false }));
 
       // Pre-load values for default labels
-      return labelKeys.filter(key => DEFAULT_KEYS.indexOf(key) > -1).map(key => this.fetchLabelValues(key));
+      return Promise.all(
+        labelKeys.filter(key => DEFAULT_KEYS.indexOf(key) > -1).map(key => this.fetchLabelValues(key))
+      ).then(values => {
+        this.logLabelFetchTs = Date.now();
```

Comment:
Race condition: `logLabelFetchTs` must be set before the requests start, otherwise a lot of requests can be started if  `refreshLogLabels` is called quickly, assuming  `logLabelFetchTs` has been set once (`Date.now() - this.logLabelFetchTs > LABEL_REFRESH_INTERVAL` with `this.logLabelFetchTs` being `undefined` masks this on the first run.

Reply:
@davkal this is fixed in 0b17d65